sdk/communication/AzureCommunicationChat/Source/Registrar/RegistrarSettings.swift

```diff
@@ -0,0 +1,56 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+
+internal enum RegistrarSettings {
+    // Node id is unused but we are required to send it
+    static let nodeId = ""
+
+    // ClientDescription Settings
+    static let appId = "AcsIos"
+    static let languageId = ""
+    static let platform = "iOS"
+    static let platformUIVersion = "3619/0.0.0.0/"
+    static let templateKey = "AcsIos.AcsNotify_Chat_3.0"
+
+    // Transport Settings
+    static let pushNotificationTransport = "APNS"
+    // Max TTL is 180 days
+    static let ttl = 15_552_000
+    static let context = ""
```

Comment:
These are all constants, but where do these values actually come from? Why are they all static? Why are there empty strings (which cannot be modified since they are `let`)?

In short, I don't understand the need for this object.

Reply:
This class contains the static values required in each registration request.
Based on the description of Registrar API ([EDFRegistrar Notebook (sharepoint.com)](https://microsoft.sharepoint.com/teams/EDFRegistrar/_layouts/15/Doc.aspx?sourcedoc=%7Bd54f197a-d828-4c78-bb08-42eb962b8211%7D&action=view&wd=target%28Home.one%7C03f29ff8-34f5-4e72-b688-7e0f89984628%2FRegistrar+API%7C53b2b5b3-3b4a-419d-8c41-69b82ec89ae7%2F%29&isSPOFile=1)),
we will need to include "nodeId", "clientDescription" and "transports" in our registration request. Most of them are static and pre-defined. Some fields are required although they have empty values - that's why you see some empty strings.

Reply:
These constants should either be default values in the `RegistrarClientDescription` initializer or, if they are truly required constants, not in the initializer at all but simply defined on the `RegistrarClientDescription`.

Reply:
OK I put them in the initializer.